services/wallet/reader.go

```diff
@@ -657,10 +651,6 @@ func (r *Reader) GetWalletToken(ctx context.Context, addresses []common.Address)
 
 	r.lastWalletTokenUpdateTimestamp.Store(time.Now().Unix())
 
-	for communityID := range communities {
-		r.communityManager.FetchCommunityMetadataAsync(communityID)
```

Comment:
How is link with what's happening UI side?

If a community is unknown to the user how we will he request it?
Also why moving that UI side?

Reply:
Yeah i got the fact that it is joining. But u need to have the data in your db.

The community model, parse all the tokens with a community id and fetch it?